app/src/main/java/org/schabi/newpipe/RouterActivity.java

```diff
@@ -257,70 +258,110 @@ private void showUnsupportedUrlDialog(final String url) {
+                preferences.getString(getString(R.string.preferred_open_action_key),
+                        getString(R.string.preferred_open_action_default)));
+
+        // Check for non-player related choices
+        if (choiceChecker.isAvailableAndSelected(
+                R.string.show_info_key,
+                R.string.download_key,
+                R.string.add_to_playlist_key)) {
+            handleChoice(choiceChecker.getSelectedChoiceKey());
+            return;
+        }
+        // Check if the choice is player related
+        if (choiceChecker.isAvailableAndSelected(
+                R.string.video_player_key,
+                R.string.background_player_key,
+                R.string.popup_player_key)) {
+
+            final String selectedChoice = choiceChecker.getSelectedChoiceKey();
+
             final boolean isExtVideoEnabled = preferences.getBoolean(
                     getString(R.string.use_external_video_player_key), false);
             final boolean isExtAudioEnabled = preferences.getBoolean(
                     getString(R.string.use_external_audio_player_key), false);
-            final boolean isVideoPlayerSelected = selectedChoiceKey.equals(videoPlayerKey)
-                    || selectedChoiceKey.equals(popupPlayerKey);
-            final boolean isAudioPlayerSelected = selectedChoiceKey.equals(backgroundPlayerKey);
-
-            if (currentLinkType != LinkType.STREAM) {
-                if (isExtAudioEnabled && isAudioPlayerSelected
-                        || isExtVideoEnabled && isVideoPlayerSelected) {
-                    Toast.makeText(this, R.string.external_player_unsupported_link_type,
-                            Toast.LENGTH_LONG).show();
-                    handleChoice(showInfoKey);
-                    return;
-                }
+            final boolean isVideoPlayerSelected =
+                    selectedChoice.equals(getString(R.string.video_player_key))
+                            || selectedChoice.equals(getString(R.string.popup_player_key));
+            final boolean isAudioPlayerSelected =
+                    selectedChoice.equals(getString(R.string.background_player_key));
+
+            if (currentLinkType != LinkType.STREAM
+                    && (isExtAudioEnabled && isAudioPlayerSelected
+                    || isExtVideoEnabled && isVideoPlayerSelected)
+            ) {
+                Toast.makeText(this, R.string.external_player_unsupported_link_type,
+                        Toast.LENGTH_LONG).show();
+                handleChoice(getString(R.string.show_info_key));
+                return;
             }
 
-            final List<StreamingService.ServiceInfo.MediaCapability> capabilities
-                    = currentService.getServiceInfo().getMediaCapabilities();
+            final List<StreamingService.ServiceInfo.MediaCapability> capabilities =
+                    currentService.getServiceInfo().getMediaCapabilities();
```

Comment:
Uhm, why isn't checkstyle complaining? The `=` should be on the newline

Reply:
Actually, from the checkstyle link you provided, the default value is `nl`, i.e. putting the operator on new line. It always has been this way in NewPipe, so I think you should stick to that (as you did e.g. for `&&` and `||` in `if`s), though this is really minor.

Reply:
> Actually, from the checkstyle link you provided, the default value is nl, i.e. putting the operator on new line.

Uhm no:
![grafik](https://user-images.githubusercontent.com/40789489/170789118-5674bfba-755d-4fb5-9e81-e65b3b89957b.png)

It only checks the tokens in the right (not left) column.


> It always has been this way in NewPipe...

Also no. Example: https://github.com/TeamNewPipe/NewPipe/blob/1daece3bee8a2955423750706199819338724f8f/app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java#L138-L147 or https://github.com/TeamNewPipe/NewPipe/blob/1daece3bee8a2955423750706199819338724f8f/app/src/main/java/org/schabi/newpipe/download/DownloadDialog.java#L131-L139

Update: I went a bit through <sub>git</sub> history and found these commits:
* https://github.com/TeamNewPipe/NewPipe/pull/3282 - https://github.com/TeamNewPipe/NewPipe/pull/3282/commits/fda5405e48f7cb1f4f4b9df89052d55c90e8b167 (looks like the origin of the ``=`` on new line)
* https://github.com/TeamNewPipe/NewPipe/pull/2907 - https://github.com/TeamNewPipe/NewPipe/commit/d8f7db4715cea3c405e11ba28cfa809d9c6447a4 (overwrites/reverts some of these changes)

So apparently someone (*cough* [wb9688](https://github.com/wb9688)) introduced his personal code "style" but didn't persist the changes inside checkstyle and also no one noticed it inside the PR...

Anyway I - and I also  think most other developers (as seen above) - favor the ``=`` after assignment style. So I would stick with that ;)

Reply:
Mmmh ok. I still don't see why all operators should be on newlines (like you did for `||` below here) but `=` should not, but anyway... If this is the standard across the industry, if you have a little spare time please open a PR to enforce such style in checkstyle